disnake/channel.py

```diff
@@ -1817,6 +1853,479 @@ class NewsChannel(TextChannel):
+            return self.__class__(state=self._state, guild=self.guild, data=payload)  # type: ignore
+
+    @utils.copy_doc(disnake.abc.GuildChannel.clone)
+    async def clone(
+        self, *, name: Optional[str] = None, reason: Optional[str] = None
+    ) -> ForumChannel:
+        return await self._clone_impl(
+            {
+                "topic": self.topic,
+                "nsfw": self.nsfw,
+                "default_auto_archive_duration": self.default_auto_archive_duration,
+            },
+            name=name,
+            reason=reason,
+        )
+
+    def get_thread(self, thread_id: int, /) -> Optional[Thread]:
+        """Returns a thread with the given ID.
+
+        Parameters
+        ----------
+        thread_id: :class:`int`
+            The ID to search for.
+
+        Returns
+        -------
+        Optional[:class:`Thread`]
+            The returned thread of ``None`` if not found.
+        """
+        return self.guild.get_thread(thread_id)
+
+    async def fetch_thread(self, thread_id: int, /) -> Thread:
+        """|coro|
+
+        Retrieves a :class:`.Thread` with the given ID.
+
+        .. note::
+
+            This method is an API call. For general usage, consider :meth:`get_thread` instead.
+
+        Raises
+        ------
+        InvalidData
+            An unknown channel type was received from Discord
+            or the guild the channel belongs to is not the same
+            as the one in this object points to.
+        HTTPException
+            Retrieving the channel failed.
+        NotFound
+            Invalid Channel ID.
+        Forbidden
+            You do not have permission to fetch this channel.#
+
+        Returns
+        -------
+        :class:`.Thread`
+            The channel from the ID.
+        """
+        return await self.guild.fetch_channel(thread_id)  # type: ignore
+
+    async def create_thread(
+        self,
+        *,
+        name: str,
+        auto_archive_duration: AnyThreadArchiveDuration = MISSING,
+        slowmode_delay: int = MISSING,
+        content: Any,
+        embed: Embed = MISSING,
+        embeds: List[Embed] = MISSING,
+        file: File = MISSING,
+        files: List[File] = MISSING,
+        stickers: Sequence[Union[GuildSticker, StickerItem]] = MISSING,
+        allowed_mentions: AllowedMentions = MISSING,
+        view: View = MISSING,
+        components: Components = MISSING,
+        reason: Optional[str] = None,
+    ) -> Thread:
+        """|coro|
+
+        Creates a thread in this forum channel.
+
+        You must have the :attr:`~Permissions.send_messages` permission to do this.
+
+        All the things that are in :meth:`TextChannel.send` are also valid here.
+
+        .. danger::
+
+            This method does two API calls, one for creating the thread and one for
+            sending the message. Please be aware of this to avoid doing a lot of
+            API calls.
+
```

Comment:
No longer true! Discord fixed the endpoint!

Reply:
3bd081fffbbbeb98223139ae0e743ee1d2ef6659